ui/embeds/stats.py

```diff
@@ -15,19 +17,29 @@ async def stats_embed(
     bot: "DiscordBot",
     leetcode_id: str,
     display_name: str,
+    display_url: bool,
     extension: StatsCardExtensions,
 ) -> tuple[discord.Embed, discord.File | None]:
-    file = await stats_card(bot, leetcode_id, extension)
+    # use a randomised filename instead of the user's leetcode_id
+    filename = ''.join(random.choices(string.ascii_letters + string.digits, k=25))
+
+    file = await stats_card(bot, leetcode_id, filename, extension, display_url)
 
     if not file:
         return error_embed(), None
 
-    embed = discord.Embed(
-        title=display_name,
-        url=f"https://leetcode.com/{leetcode_id}",
-        colour=discord.Colour.orange(),
-    )
-    embed.set_image(url=f"attachment://{leetcode_id}.png")
+    if display_url:
+        embed = discord.Embed(
+            title=display_name,
+            url=f"https://leetcode.com/{leetcode_id}",
+            colour=discord.Colour.orange(),
+        )
+    else:
+        embed = discord.Embed(
+            title=display_name,
+            colour=discord.Colour.orange(),
+        )
```

Comment:
Rather than handling two conditions, let's just set `url` to the url if display_url, else set it to None. I think that should work, but first check the default type of that param.